packages/luigis-box/src/Model/Batch/LuigisBoxBatchLoader.php

```diff
@@ -0,0 +1,202 @@
+            }
+
+            if ($type === 'brand') {
+                $mappedDataOfCurrentType = $this->mapBranchData($luigisBoxResults[$type]);
+                self::$totalsByType[$type] = count($mappedDataOfCurrentType);
+            }
+
+            $mappedData[] = $mappedDataOfCurrentType;
+        }
+
+        return $mappedData;
+    }
+
+    /**
+     * @param \Shopsys\LuigisBoxBundle\Component\LuigisBox\LuigisBoxResult $luigisBoxResult
+     * @param int $limit
+     * @return array
+     */
+    protected function mapProductData(LuigisBoxResult $luigisBoxResult, int $limit): array
+    {
+        $filterQuery = $this->filterQueryFactory->createSellableProductsByProductIdsFilter($luigisBoxResult->getIds(), $limit);
+
+        return $this->productElasticsearchRepository->getSortedProductsResultByFilterQuery($filterQuery)->getHits();
+    }
+
+    /**
+     * @param \Shopsys\LuigisBoxBundle\Component\LuigisBox\LuigisBoxResult $luigisBoxResult
+     * @return \Shopsys\FrameworkBundle\Model\Category\Category[]
+     */
+    protected function mapCategoryData(LuigisBoxResult $luigisBoxResult): array
+    {
+        $categoryArray = $this->categoryFacade->getVisibleCategoriesByIds([$luigisBoxResult->getIds()], $this->domain->getCurrentDomainConfig());
+
+        return reset($categoryArray);
+    }
+
+    /**
+     * @param \Shopsys\LuigisBoxBundle\Component\LuigisBox\LuigisBoxResult $luigisBoxResult
+     * @return array
+     */
+    protected function mapArticleData(LuigisBoxResult $luigisBoxResult): array
+    {
+        if (count($luigisBoxResult->getIdsWithPrefix()) === 0) {
+            return [];
+        }
+
+        $idsByType = [];
+
+        foreach ($luigisBoxResult->getIdsWithPrefix() as $idWithPrefix) {
+            [$type, $id] = explode('-', $idWithPrefix);
+            $idsByType[$type][] = $id;
+        }
+
+        return $this->combinedArticleElasticsearchFacade->getArticlesByIds(
+            $idsByType,
+            $this->domain->getId(),
+        );
+    }
+
+    /**
+     * @param \Shopsys\LuigisBoxBundle\Component\LuigisBox\LuigisBoxResult $luigisBoxResult
+     * @return \Shopsys\FrameworkBundle\Model\Product\Brand\Brand[]
+     */
+    protected function mapBranchData(LuigisBoxResult $luigisBoxResult): array
+    {
+        return $this->brandFacade->getBrandsByIds($luigisBoxResult->getIds());
+    }
+
+    /**
+     * @param array $luigisBoxBatchLoadData
+     */
+    protected function validateProductSearchIsPresent(array $luigisBoxBatchLoadData): void
+    {
+        $productSearchIsPresent = false;
+
+        foreach ($luigisBoxBatchLoadData as $luigisBoxBatchLoadDataItem) {
+            if ($luigisBoxBatchLoadDataItem->getType() === LuigisBoxClient::MAIN_TYPE) {
+                $productSearchIsPresent = true;
+
+                break;
+            }
+        }
+
+        if (!$productSearchIsPresent) {
+            throw new ProductSearchIsMandatoryForAllLuigisBoxSearchesException();
```

Comment:
When I call this query with luigisbox enabled, I get 500 error. I would expect a nice graphql error explaining what did I do wrong

```{
  categoriesSearch(search:"ele"){
  edges{
    node{
      name
      seoH1
    }
  }
}}
```

Reply:
I have added user error